pkg/pipeline/transform/transform_network.go

```diff
@@ -152,6 +165,17 @@ func (n *Network) Transform(inputEntry config.GenericMap) (config.GenericMap, bo
 	return outputEntry, true
 }
 
+func (n *Network) categorizeIP(ip net.IP) string {
+	for _, subnetCat := range n.categories {
+		for _, cidr := range subnetCat.cidrs {
+			if cidr.Contains(ip) {
+				return subnetCat.name
+			}
+		}
+	}
+	return ""
+}
+
```

Comment:
I'm afraid this exhaustive search could increase noticeably the CPU usage. I'd add an lru cache implementation (like [this](https://github.com/netobserv/flowlogs-pipeline/blob/main/pkg/pipeline/utils/timed_cache.go) or [this](https://github.com/mariomac/guara/blob/main/pkg/cache/lru.go)) and check there the IP first.
In pseudo-golang:
```go
func (n *Network) categorizeIP(ip net.IP) string {
	if category, ok := n.cache.Get(ip); ok {
		return category
	}
	for _, subnetCat := range n.categories {
		for _, cidr := range subnetCat.cidrs {
			if cidr.Contains(ip) {
				n.cache.Put(ip, subnetCat.name)
				return subnetCat.name
			}
		}
	}
	n.cache.Put(ip, "")
	return ""
}
```

Reply:
Yes I have the same concern and wanted to point out in the documentation (when this is implemented on operator side) that this could be disabled to save processor's resource.
But +1 about implementing a cache right now

Reply:
done
I took the opportunity to update a little bit `TimedCache` in order to use go's time API + a couple of small stuffs